core/src/test/java/com/nextdoor/bender/handler/BaseHandlerTest.java

```diff
@@ -160,6 +181,33 @@ public void testEndToEnd() throws HandlerException {
     assertEquals("bar", BufferedTransporter.output.get(1));
   }
 
+  @Test
+  public void testIeventGetsMetadata() throws Throwable {
+    BaseHandler.CONFIG_FILE = "/config/handler_config.json";
+
+    List<DummyEvent> events = new ArrayList<DummyEvent>(2);
+    events.add(new DummyEvent("foo", 0));
+    events.add(new DummyEvent("bar", 0));
+
+    TestContext context = new TestContext();
+    context.setInvokedFunctionArn("arn:aws:lambda:us-east-1:123:function:test:tag");
+    handler.init(context);
+    IpcSenderService spyIpc = spy(handler.getIpcService());
+
+    ArgumentCaptor<InternalEvent> captor = ArgumentCaptor.forClass(InternalEvent.class);
+    TransportFactory tf = spy(handler.getIpcService().getTransportFactory());
+    BufferedTransporter mockTransport = mock(BufferedTransporter.class);
+    when(tf.newInstance()).thenReturn(mockTransport);
+    spyIpc.setTransportFactory(tf);
+
+    handler.setIpcService(spyIpc);
+
+    handler.handler(events, context);
+
+    verify(spyIpc, times(2)).add(captor.capture());
+    captor.getAllValues().get(0).getMetadata();
```

Comment:
Add an `assertEquals` here for the metadata of both events. eg
```
assertEquals(expected, captor.getAllValues().get(0).getMetadata());
assertEquals(expected, captor.getAllValues().get(1).getMetadata());
```

See note about adding equals and hashcode.